### DISCUSSION TITLE
Data List: Groups

### DISCUSSION THREAD

**Discussion by lindeberg** [leekelleher/umbraco-contentment#90] [Idea]
It would be useful to be able to [group ](https://www.w3schools.com/tags/tag_optgroup.asp) the dropdown list items.

I am mostly thinking about when using a [custom data source](https://github.com/leekelleher/umbraco-contentment/blob/develop/docs/editors/data-list.md#extending-with-your-own-custom-data-source).

### Describe the solution you'd like
I would like to have a settable 'Group'-property on the ['DataListItem'-class](https://github.com/leekelleher/umbraco-contentment/blob/develop/src/Umbraco.Community.Contentment/DataEditors/DataList/DataListItem.cs)

**Comment by leekelleher** [comment]
@lindeberg Thank you for the suggestion. 💡 

I'd considered an option for grouping data-items, but it started to add complexity to the other list-editors. Although, that was about a year ago, so I'm happy to give it some more thought.

**Comment by leekelleher** [comment]
Update on this, I've added a branch for my prototype for adding data-item grouping.
The branch is [`dev/wip/data-list-groups`](https://github.com/leekelleher/umbraco-contentment/tree/dev/wip/data-list-groups). Commit 34d1f68c1ca992e3dd2622550216174d36be5f2a has the significant code amends.

Initially the feature request may seem to be _"add a `Group` property to the `DataListItem` class",_ but I would immediate be asked why the built-in list editors didn't support groupings, so I believe it's all part and parcel of the feature. A holistic feature, so to speak. 😃 

From my prototyping, both Dropdown List and Radio Button List look fine, I'm less happy with Checkbox List (most likely needs more TLC), and I hadn't looked at how it would work with the Item Picker editor.

While this feature is not high on my current priorities, I am still interested in (or others) exploring this in future. This branch should provide a good foundation.

**Comment by leekelleher** [comment]
Linking for prosperity... Ulrich Ebsen's example of his custom button list, grouped with headings.
<https://twitter.com/ecoredk/status/1331513128618745856>

**Comment by leekelleher** [comment]
Moved this topic to a discussion. I'm growing fonder about the grouping idea. I'll use this thread for idea discussion, brainstorming, etc.
